### PR TITLE
Minor clean up of the data code

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Improves the docstrings for DataPHA, adds tests for several corner-cases, and makes some minor code improvements to the code.

# Details

This was found when looking at #1160, #1427 and related, but have been pulled out as it would be nice to get the functionality tested and the minor code improvements added before I spend a lot of time getting stuck on the general topics.

The code changes are that we now use `super()` rather than explicitly calling the parent class in the `__init__` method. We can not do this for all classes in the data/astro.data hierarchy as we don't always have a "clean" hierarchy (and is not something I am planning on addressing here).

A number of tests are labelled as `xfail` rather than just accepting the current behavior. We could change this, but I plan to add PRs to address these tests in the 4.14.1 time frame so they hopefully won't last long.

### CODE REVIEW

**Review by hamogu — APPROVED** [review]
Except for a typo in the comment in the docs...

File: sherpa/astro/tests/test_astro_data2.py [hunk 1]
```diff
@@ -1210,11 +1157,476 @@ def test_pha_grouping_changed_filter_1160(make_test_pha):
...
+
+    pha.grouping = [1, -1, 1, -1]
+    assert not pha.grouped
+    pha.grouped = True
+    d1 = pha.get_dep(filter=True)
+    assert d1 == pytest.approx([3, 3])
+
+    # Can we remove the grouping column?
+    pha.grouping = None
+    # This thinks that pha.grouped is sill set
```
[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/astro/tests/test_astro_data2.py:1187]: Yes, but this test is marked xfail, so this comment actually points to why the test fails (I've just found it useful to point out why a test is marked as xfail, but perhaps there should be a more-explicit note about it).

File: sherpa/astro/data.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-#  Copyright (C) 2008, 2015, 2016, 2017, 2018, 2019, 2020, 2021
+#  Copyright (C) 2008, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022
```
[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/astro/data.py:2]: It might have been thanks to a script of yours
